rp-binary-info/src/macros.rs

```diff
@@ -166,4 +178,20 @@ macro_rules! rp_pico_board {
     };
 }
 
+/// Generate a static item containing the binary end address, and return its
+/// [`EntryAddr`](super::EntryAddr). The argument should be a symbol provided
+/// by the linker script that is located at the end of the binary.
+#[macro_export]
+macro_rules! rp_binary_end {
+    ($ptr:ident) => {{
+        $crate::pointer!(
+            $crate::consts::TAG_RASPBERRY_PI,
+            $crate::consts::ID_RP_BINARY_END,
+            // `unsafe` only needed because MSRV does not yet
+            // contain https://github.com/rust-lang/rust/pull/125834
+            unsafe { core::ptr::addr_of!($ptr).cast() }
```

Comment:
Will this trip 'unneeded unsafe' warnings?

Reply:
No, not even on nightly.

I guess this is one of the lints that get ignored if they trigger inside an external macro. But I didn't find documentation about this feature. So all I know is that the binary_info_demo compiled without warnings on all compilers I tried.